### PR TITLE
GitHub update

### PR DESCRIPTION
**Pull request by Alkarex**
https://github.com/FreshRSS/FreshRSS/issues/907
https://github.com/FreshRSS/update.freshrss.org
Work in progress

### CODE REVIEW

File: app/update/update_util.php [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+		}
+	}
+
+	$index = $dir . '/index.html';
+	$data = <<<EOF
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en-GB" lang="en-GB">
+<head>
+<meta charset="UTF-8" />
+<meta http-equiv="Refresh" content="0; url=/" />
```
Comment (Frenzie) [inline app/update/update_util.php]: Is this root `/` quite what is wanted here?

Comment (Alkarex) [reply app/update/update_util.php]: This is some code from Marien, directly from the current update server.
This part is generating an index.html. I have a habit of putting an index.html file in directories that should not be browsed (even when there are other rules such as .htaccess, etc.). The redirect address is not very important, as it is not supposed to be used (except during manual / accidental browsing).

**Comment by kevinpapst** [comment]
This might not be directly related to this PR, but its related to the update mechanism: 
I would kindly ask for a more thoroughly permission check (updateController ~line 80). When I crashed my FreshRSS installation, that was due to a stupid permission problem. Some files were writable, some not. The current is_writable() check only tests the main directory and does not act friendly if some files are not writable. Adding a simple recursive file and directory check would likely solve the issue. Yes, it should not be a FreshRSS task to handle file permissions properly, but it should warn the user if the update could potentially fail.

**Comment by Alkarex** [comment]
I can see I am getting in late for this PR. Any help welcome :-)

**Comment by Alkarex** [comment]
Damn, I have broken my PR :-(

**Comment by Alkarex** [comment]
Replaced by https://github.com/FreshRSS/FreshRSS/pull/1760
Sorry for the wrong manipulation.

**Pull request closed without merging**